src/bp-members/classes/class-bp-members-notice.php

```diff
@@ -0,0 +1,758 @@
+
+		if ( ! $wpdb->query( $sql ) ) {
+			return false;
+
+			// Remove all corresponding notice metadata.
+		} else {
+			bp_notices_delete_meta( $this->id );
+		}
+
+		/**
+		 * Please do not use this filter anymore.
+		 *
+		 * @since 2.8.0
+		 * @deprecated 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice Current instance of the notice being saved. Passed by reference.
+		 */
+		do_action_deprecated( 'messages_notice_after_delete', array( $this ), '15.0.0', 'bp_members_notice_after_delete' );
+
+		/**
+		 * Fires after the notice item has been deleted.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param BP_Members_Notice $notice Current instance of the notice item being deleted.
+		 */
+		do_action( 'bp_members_notice_after_delete', $this );
+
+		return true;
+	}
+
+	/** Static Methods ********************************************************/
+
+	/**
+	 * Pulls up a list of notices.
+	 *
+	 * To get all notices, pass a value of -1 to pag_num.
+	 *
+	 * @global wpdb $wpdb WordPress database object.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @param array $args {
+	 *     An array of arguments.
+	 *
+	 *     @type integer      $user_id          The user ID to get notices for. Defaults to `0`.
+	 *     @type integer      $pag_num          Number of notices per page. Defaults to `20`.
+	 *     @type integer      $pag_page         The page number to retrieve.  Defaults to `1`.
+	 *     @type boolean      $dismissed        `true` to get dismissed notices. False otherwise. Defaults to `false`.
+	 *     @type array        $meta_query       An array describing the Meta Query to perform. Defaults to `[]`.
+	 *     @type string       $fields           Use 'ids' to only get Notice IDs, 'all' otherwise. Defaults to 'all'.
+	 *     @type string       $type             Use 'active' to only get active notices, 'all' to get all notices or
+	 *                                          'inactive' to get inactive notices. Defaults to 'active'.
+	 *     @type array        $exclude          An array of notice IDs to exclude from results. Defaults to `[]`.
+	 *     @type array        $target__in       An array of targeted audiences to include in results. Possible targets
+	 *                                          are 'community', 'contributors', 'admins'. Defaults to `[]`.
+	 *     @type null|integer $priority         Use `1` to get high priority notices, `2` to get regular ones or `3` to
+	 *                                          get low ones. `0` is restriced to BuddyPress usage: please don't use it.
+	 *                                          Defaults to `null`.
+	 *     @type boolean      $count_total_only `true` to only get the total number of notices, `false` otherwise.
+	 *                                          Defaults to `false`.
+	 * }
+	 * @return array|integer|null List of notices, total number of notices for a count only query or null if nothing was found.
```

Comment:
Most of our `::get`, returns the objects (or ids only) and the total number of matches objects at the same time. I think maybe to avoid a second query. Not all classes support this though.

Reply:
Most of these `::get` are performing 2 queries: one to list objects, the other to get total, I believe using it this way is avoiding making 2 queries when we only need one 😄